.circleci/create_circleci_config.py

```diff
@@ -401,6 +409,36 @@ def job_name(self):
     tests_to_run="tests/repo_utils",
 )
 
+doc_test_job = CircleCIJob(
+    "pr_documentation_tests",
+    additional_env={"TRANSFORMERS_VERBOSITY":"error", "DATASETS_VERBOSITY":"error", "SKIP_CUDA_DOCTEST": "1"},
+    install_steps=[
+        "sudo apt-get -y update && sudo apt-get install -y libsndfile1-dev espeak-ng time",
+        "pip install --upgrade pip",
+        "pip install -e .[dev]",
+        "pip install git+https://github.com/huggingface/accelerate",
+        "pip install --upgrade pytest pytest-sugar",
+        "find -name __pycache__ -delete",
+        "find . -name \*.pyc -delete",
+        {
+            "name": "Get files to test",
+            "command":
+                "git remote add upstream https://github.com/huggingface/transformers.git  && git fetch upstream \n"
+                "git diff --name-only --relative --diff-filter=AMR refs/remotes/upstream/main...HEAD | grep -E '\.(py|mdx)$' | grep -Ev '^\..*|/\.' | grep -Ev '__' > pr_documentation_tests.txt"
+        },
+        {
+            "name": "List files beings tested : `pr_documentation_tests.txt`",
+            "command":
+                "cat pr_documentation_tests.txt"
+        },
+    ],
+    tests_to_run="$(cat pr_documentation_tests.txt)",
+    pytest_options={"-doctest-modules":None, "doctest-glob":"*.mdx","dist":"loadfile", "rvsA":None},
```

Comment:
`-doctest-modules` seems incorrect and should be `doctest-modules`

Reply:
also, the format should be better: spaces after `:` or `,`

Reply:
`doctest-modules` alone doesn’t work, since it’s only adding a single `-`

Reply:
> `doctest-modules` alone doesn’t work, since it’s only adding a single `-`

For such *unusual* situation, I would recommend to leave some comments in the code (and if not, in the PR page) to facilitate the review as well as for future HF developers.

Reply:
I'm not sure I follow, why is it adding only a single `-`?  I agree with @ydshieh, this is non-obvious and to someone coming cold to the code I think confusing without further context.

Reply:
@amyeroberts I think it is because we use `--doctest-modules` without argument. But in `create_circleci_config.py` we have

```python
pytest_flags = [f"--{key}={value}" if value is not None else f"-{key}" for key, value in all_options.items()]
```
So we need put `-doctest-modules` and the above line will change it to `--doctest-modules` (that is what is required)

This is exceptional however, and we need to make it clear in some way :-)

Reply:
In this case, I'd say we should update this logic as it's not properly handling command line options :) e.g. something like this: 

```
PYTEST_BOOL_FLAG = ['doctest-modules',]

... 
pytest_flags = []
for key, value in all_options.items():
    if key in PYTEST_BOOL_FLAG:
    	flag = f"--{key}"
    elif value is not None:
    	flag = f"--{key}={value}"
    else:
    	flag = f"-{key}
    pytest_flags.append(flag)
```

Reply:
I do agree :-)